dags/openshift_nightlies/dag.py

```diff
@@ -213,13 +213,27 @@ def build(self):
         mgmt_installer = self._get_openshift_installer()
         hosted_installer = self._get_hypershift_openshift_installer()
         install_mgmt_cluster = mgmt_installer.get_install_task()
+        connect_to_platform = self._get_platform_connector().get_task()
 
         rosa_post_installation = self._get_rosa_postinstall_setup()._get_rosa_postinstallation()
 
-        install_hosted_cluster = hosted_installer.get_hosted_install_task()
-        for c_id, cluster in install_hosted_cluster:
-            benchmark = self._add_benchmarks(task_group=c_id)
-            install_mgmt_cluster >> rosa_post_installation >> cluster >> benchmark
+        if self.config.cleanup_on_success:
+            install_hosted_cluster = hosted_installer.get_hosted_install_task()
+            cleanup_mgmt_cluster = mgmt_installer.get_cleanup_task()
+            cleanup_hosted_cluster = hosted_installer.get_cleanup_task()
+            for c_id, cluster in install_hosted_cluster:
+                benchmark = self._add_benchmarks(task_group=c_id)
+                hc_connect_to_platform = self._get_hc_platform_connector(task_group=c_id).get_task()            
+                install_mgmt_cluster >> rosa_post_installation >> connect_to_platform
+                connect_to_platform >> cluster >> [hc_connect_to_platform, benchmark]
```

Comment:
can we use "hosted_cluster" or "install_hc" instead of "cluster"? will reduce confusion. also @mukrishn could you trigger a playground for this PR, where number_of_hosted_cluster is set to 2, best to test the changes on a playground first